Support uni-directional UART for cc26x0-cc13x0

arurke opened the pull request:
Removes the requirement for both UART TX and RX to be set, thus adding support for boards which only have one or the other. (Same PR as https://github.com/contiki-os/contiki/pull/2259)

g-oikonomou commented:
Can we have a single function `usable()` that accepts `BOARD_IOID_UART_xx` as an argument and compares with `IOID_UNUSED`? Not sure what the argument data type would be, off the top of my head I think `uint32_t`?

arurke commented:
> Can we have a single function usable() that accepts BOARD_IOID_UART_xx as an argument and compares with IOID_UNUSED?

Sure. Updated and added some minor cleaning.

> I think uint32_t

That's correct

g-oikonomou commented on arch/cpu/cc26x0-cc13x0/dev/cc26xx-uart.c:
```diff
@@ -297,12 +294,12 @@ cc26xx_uart_init()
   }
 
-  if(accessible() == false) {
+  if(accessible()) {
```
That seems wrong

arurke replied:
🤦  Quite impressive. Fixed.

g-oikonomou commented on arch/cpu/cc26x0-cc13x0/dev/cc26xx-uart.c:
```diff
@@ -346,13 +343,13 @@ cc26xx_uart_set_input(int (*input)(unsigned char c))
 
   /* If the UART is not accessible, it is not busy */
-  if(accessible() == false) {
+  if(accessible()) {
```
Ditto

arurke replied:
🤦  Quite impressive. Fixed.

g-oikonomou commented:
ReAdy with that?
